usecases/schema/handler.go

```diff
@@ -102,38 +112,60 @@ func NewHandler(
 		scaleOut:                scaleoutManager,
 	}
 
-	// TODO-RAFT
-	// m.scaleOut.SetSchemaManager(m)
+	handler.scaleOut.SetSchemaManager(metaReader)
 
-	return m, nil
+	return handler, nil
 }
 
 // GetSchema retrieves a locally cached copy of the schema
-func (m *Handler) GetSchema(principal *models.Principal) (schema.Schema, error) {
-	err := m.Authorizer.Authorize(principal, "list", "schema/*")
+func (h *Handler) GetSchema(principal *models.Principal) (schema.Schema, error) {
+	err := h.Authorizer.Authorize(principal, "list", "schema/*")
 	if err != nil {
 		return schema.Schema{}, err
 	}
 
-	return m.getSchema(), nil
+	return h.getSchema(), nil
 }
 
 // GetSchemaSkipAuth can never be used as a response to a user request as it
 // could leak the schema to an unauthorized user, is intended to be used for
 // non-user triggered processes, such as regular updates / maintenance / etc
-func (m *Handler) GetSchemaSkipAuth() schema.Schema { return m.getSchema() }
+func (h *Handler) GetSchemaSkipAuth() schema.Schema { return h.getSchema() }
 
-func (m *Handler) getSchema() schema.Schema {
-	s := m.metaReader.ReadOnlySchema()
+func (h *Handler) getSchema() schema.Schema {
+	s := h.metaReader.ReadOnlySchema()
 	return schema.Schema{
 		Objects: &s,
 	}
 }
 
-func (m *Handler) Nodes() []string {
-	return m.clusterState.AllNames()
+func (h *Handler) Nodes() []string {
+	return h.clusterState.AllNames()
+}
+
+func (h *Handler) NodeName() string {
+	return h.clusterState.LocalName()
 }
 
-func (m *Handler) NodeName() string {
-	return m.clusterState.LocalName()
+func (h *Handler) UpdateShardStatus(ctx context.Context,
```

Comment:
I'm sure whether we should make 'UpdateShardStatus()' part of the Handler. It doesn't seem to be directly related to schema updates, and, as a result, consensus may not be necessary in this context. 

As part of interface segregation, it might be a good idea to create a new class specifically dedicated to handling this aspect

Reply:
I would see it as part of the schema information that should be propagated and require consensus. Right now the side-effect of a shard being set active or inactive is isolated to a single node, but in the future this could also mean it is being offloaded to cloud storage. I assume that is something that we would want the whole cluster to have consensus on?

Reply:
This method and `ShardsStatus` handle the `/v1/schema/{className}/shards` endpoint, so currently it is expected that the schema manger (and therefore Handler) would be responsible here.

If consensus is necessary here, it would make sense to keep this logic here. If not, I agree that a different class can handle this